src/Observations.jl

```diff
@@ -473,8 +501,10 @@ function combine_observations(obs_vec::AV) where {AV <: AbstractVector}
     nnew2 = [convert(T, n) for n in nnew]
     T = promote_type((typeof(i) for i in inew)...)
     inew2 = [convert(T, i) for i in inew]
+    T = promote_type((typeof(i) for i in mnew)...)
```

Comment:
```suggestion
try
    T = promote_type((typeof(i) for i in mnew)...)
catch
    throw(ArgumentError("Incompatible metadata types detected: $(unique(typeof.(mnew)))"))
end
```
Since metadata can have heterogeneous types, this could fail easily. Could we add an informative warning like this one?

Reply:
This is a good point, let's just scrap it.